### PR TITLE
 `0.2.0` release pre-validation

### PR DESCRIPTION
**Pull request by zeroZshadow**
#### Description
I've made all the changes listed in https://github.com/spatialos/gdk-for-unity/wiki/Release-Process and am now opening a PR to pull this branch into develop as per that process.

#### Tests
- [x] release QA
	- [x] connect windows client
	- [x] connect mac client
	- [x] connect ios client
	- [x] connect android client
- [x] windows editor validation
- [x] mac editor validation

#### Documentation
Updated changelog

#### Primary reviewers
@oblm

### CODE REVIEW

**Comment by zeroZshadow** [comment]
Got feedback on changelog from Laura, will need to do a good amount of changes

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,17 +17,17 @@
...
-    - There is no limit on the number of `GameObject` instances that can be linked to a SpatialOS entity. However the same component type can not be added more than once.
-    - Deleting a linked `GameObject` will unlink it from the SpatialOS entity automatically.
+    - All linked `GameObject` instances are still unlinked before this is called, however it is now your responsibility to track if a `GameObject` was created when the entity was added.
+    - You should now call `linker.LinkGameObjectToSpatialOSEntity()` to link the `GameObject` to the SpatialOS entity.
+    - You should also pass-in a list of `ComponentType` to `LinkGameObjectToSpatialOSEntity` which you wish to be copied from the `GameObject` to the ECS entity associated with the `GameObject`.
+        - Note that for the Transform Synchronization feature module to work correctly, you need to set up a linked Transform Component on your GameObject. You also need to link any Rigidbody Component on your GameObject.
+    - There is no limit on the number of GameObject instances that you can link to a SpatialOS entity. However, you cannot add the same component type more than once.
```
Comment (paulbalaji) [inline CHANGELOG.md]:
```suggestion
    - There is no limit on the number of GameObject instances that you can link to a SpatialOS entity. However, you cannot add a component type to a linked GameObject instance more than once.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -51,15 +53,15 @@
...
 - Changed the format of the BuildConfiguration asset. Please recreate, or copy it from `workers/unity/Playground/Assets/Config/BuildConfiguration.asset`.
 - Building workers will not change the active build target anymore. The build target will be set back to whatever was set before starting the build process.
 
 ### Fixed
 
-- Fixed a bug where running `SpatialOS -> Generate code` would always regenerate code, even if no files had changed.
-- Fixed a bug where building all workers in our sample projects would fail, if you have Android build support installed, but didn't set the path to the Android SDK.
+- Fixed a bug where, from the SpatialOS menu in the Unity Editor, running **SpatialOS ** > **Generate code** would always regenerate code, even if no files had changed.
+- Fixed a bug where building all workers in our sample projects would fail if you have Android build support installed but didn't set the path to the Android SDK.
 - Fixed a bug where some prefabs would not be processed correctly, causing NullReferenceExceptions in OnEnable.
```
Comment (paulbalaji) [inline CHANGELOG.md]:
```suggestion
- Fixed a bug where some prefabs would not be processed correctly, causing `NullReferenceExceptions` in `OnEnable`.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -72,11 +74,12 @@
 - Re-added tests for Reader/Writer injection criteria and Monobehaviour enabling.
```
Comment (paulbalaji) [inline CHANGELOG.md]:
```suggestion
- Re-added tests for Reader/Writer injection criteria and MonoBehaviour enabling.
```

**Pull request merged**